chcemvediet/apps/inforequests/tests/test_models/test_action.py

```diff
@@ -0,0 +1,918 @@
+                (Action.TYPES.REMANDMENT,             naive_date(u'2010-10-20'), Deadline.TYPES.OBLIGEE_DEADLINE, dict()),  # 8 WD since legal_date advanced by 4 WD
+                (Action.TYPES.ADVANCED_REQUEST,       naive_date(u'2010-10-20'), Deadline.TYPES.OBLIGEE_DEADLINE, dict()),  # 8 WD since legal_date advanced by 4 WD
+                (Action.TYPES.EXPIRATION,             naive_date(u'2010-10-19'), Deadline.TYPES.APPLICANT_DEADLINE, dict()),  # 15 CD since legal_date
+                (Action.TYPES.APPEAL_EXPIRATION,      None, None, dict()),
+                )
+        # Make sure we are testing all defined action types
+        tested_action_types = set(a for a, _, _, _ in tests)
+        defined_action_types = Action.TYPES._inverse.keys()
+        self.assertItemsEqual(tested_action_types, defined_action_types)
+
+        for action_type, deadline_date, deadline_type, extra_args in tests:
+            previous = self._create_action(type=Action.TYPES.REQUEST, delivered_date=delivered_date)  # Deadline for CONFIRMATION, EXTENSION
+            action = self._create_action(type=action_type, delivered_date=delivered_date, legal_date=legal_date, **extra_args)
+            self.assertEqual(action.deadline and action.deadline.deadline_date, deadline_date)
+            self.assertEqual(action.deadline and action.deadline.type, deadline_type)
+
+    def test_deadline_property_ignores_extension_for_non_extension_action_types(self):
+        tests = (
+                (Action.TYPES.REQUEST,                dict()),
+                (Action.TYPES.CLARIFICATION_RESPONSE, dict()),
+                (Action.TYPES.APPEAL,                 dict()),
+                (Action.TYPES.CONFIRMATION,           dict()),
+                (Action.TYPES.ADVANCEMENT,            dict()),
+                (Action.TYPES.CLARIFICATION_REQUEST,  dict()),
+                (Action.TYPES.DISCLOSURE,             dict(disclosure_level=Action.DISCLOSURE_LEVELS.NONE)),
+                (Action.TYPES.DISCLOSURE,             dict(disclosure_level=Action.DISCLOSURE_LEVELS.PARTIAL)),
+                (Action.TYPES.DISCLOSURE,             dict(disclosure_level=Action.DISCLOSURE_LEVELS.FULL)),
+                (Action.TYPES.REFUSAL,                dict()),
+                (Action.TYPES.AFFIRMATION,            dict()),
+                (Action.TYPES.REVERSION,              dict()),
+                (Action.TYPES.REMANDMENT,             dict()),
+                (Action.TYPES.ADVANCED_REQUEST,       dict()),
+                (Action.TYPES.EXPIRATION,             dict()),
+                (Action.TYPES.APPEAL_EXPIRATION,      dict()),
+                )
+        # Make sure we are testing all defined action types (except EXTENSION)
+        tested_action_types = set(a for a, _ in tests)
+        defined_action_types = Action.TYPES._inverse.keys()
+        defined_action_types.remove(Action.TYPES.EXTENSION)
+        self.assertItemsEqual(tested_action_types, defined_action_types)
```

Comment:
@viliambalaz IMHO skor nieco ako

```
defined_action_types = [a for a in Action.TYPES._inverse if a != Action.TYPES.EXTENSION]
```
V pythone sa zvykne davat prednost deklarativnym comprehension pred imperativnym odoberanim z vysledku.

Reply:
:ok: pouzil som aj `.keys()` Funguje to rovnako, len som chcel aby sme boli jednotni pri definovani premennej `defined_action_types` v inych testoch.